accel/tcg/log_instr.c

```diff
@@ -1081,6 +1097,361 @@ void qemu_log_instr_extra(CPUArchState *env, const char *msg, ...)
     va_end(va);
 }
 
+/*
+ *  A printf that takes an array of argments unioned of all possible argument
+ * types. Because we cant edit a VA_LIST, and we don't want the exponential blow
+ * up of handling all combinations of types, we will bounce individual string
+ * sections split in the fmt string to another buffer, then switch on all
+ * possible types.
+ */
+static void g_string_append_printf_union_args(GString *string, const char *fmt,
+                                              qemu_log_arg_t *args)
+{
+
+/* So Clang will not complain about the non-literal format. */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#pragma GCC diagnostic ignored "-Wformat-security"
+
+#define BOUNCE_BUF_SIZE 256
+/*
+ *  A safe amount under the maximum size. An (illegally) wrong format will cause
+ * the assert to be hit, but is a bug on the callers part.
+ */
+#define BOUNCE_BUF_FLUSH_SIZE (BOUNCE_BUF_SIZE - 10)
```

Comment:
Only used once, I don't think a macro is necessary, just do `if (i >= sizeof(bounce_buf) - 10)`?